ravenframework/Metrics/metrics/StatsTestsUtils.py

```diff
@@ -0,0 +1,204 @@
+  """
+  interSegs = []
+  for i in range(interval):
+    interSegs.append(array[i::interval])
+  return np.array(interSegs)
+
+def sumSegments(array):
+  """
+    This method comes segments arrays built using the buildSegements method using a KS-test.
+    @In array1, an array-like of the historical data.
+    @In array2, an array-like of the synthetic data.
+    @In interval, int setting the length of the segment length desired
+    @ Out, a Pandas DataFrame containing the statistic and the p-value of the KS-test.
+  """
+  sum_segs = []
+  for seg in array:
+    sum_segs.append(sum(seg))
+  return np.array(sum_segs)
+
+def compare_segments_ks(array1, array2, segmentLength: int):
+  """
+  This method comes segments arrays built using the buildSegements method using a KS-test.
+    @In array1, an array-like of the historical data.
+    @In array2, an array-like of the synthetic data.
+    @In interval, int setting the length of the segment length desired
+    @ Out, a Pandas DataFrame containing the statistic and the p-value of the KS-test.
+  """
+  results = []
+  cols = ['segment','statistic', 'p-value']
+  segments1 = buildSegments(array1, segmentLength)
+  segments2 = buildSegments(array2, segmentLength)
+  if len(segments1) != len(segments2) or len(segments1[0]) != len(segments2[0]):
+    print('Warning you are attempting to compare two arrays of uneven length or with uneven length components')
+    return
+  for i in range(len(segments1)):
+    results.append(i, stats.ks_2samp(segments1[i], segments2[i]))
+  return pd.DataFrame(results, columns=cols)
+
+def compare_segments_f(array1, array2, segmentLength: int):
+  """
+    This method comes segments arrays built using the buildSegements method using a KS-test.
+    @In array1, an array-like of the historical data.
+    @In array2, an array-like of the synthetic data.
+    @In interval, int setting the length of the segment length desired
+    @ Out, a Pandas DataFrame containing the statistic and the p-value of the KS-test.
+  """
+  results = []
+  cols = ['segment','statistic', 'p-value']
+  segments1 = buildSegments(array1, segmentLength)
+  segments2 = buildSegments(array2, segmentLength)
+  if len(segments1) != len(segments2) or len(segments1[0]) != len(segments2[0]):
+    print('Warning you are attempting to compare two arrays of uneven length or with uneven length components')
+    return
+  for i in range(len(segments1)):
+    results.append(i, stats.f_oneway(segments1[i], segments2[i]))
+  return pd.DataFrame(results, columns=cols)
+
+def compare_segments_chi(array1, array2, segmentLength: int):
+  """
+    This method comes segments arrays built using the buildSegements method using a KS-test.
+    @In array1, an array-like of the historical data.
+    @In array2, an array-like of the synthetic data.
+    @In interval, int setting the length of the segment length desired
+    @ Out, a Pandas DataFrame containing the statistic and the p-value of the KS-test.
```

Comment:
names/types

Reply:
I think the name/types need to be fixed across these methods, so I won't note all of them from here forward.